src/rust/engine/process_execution/src/named_caches.rs

```diff
@@ -18,10 +18,9 @@ pub struct CacheName(String);
 
 impl CacheName {
   pub fn new(name: String) -> Result<CacheName, String> {
-    if name
-      .chars()
-      .all(|c| (c.is_ascii_alphanumeric() && c.is_ascii_lowercase()) || c == '_')
-    {
+    if name.chars().all(|c| {
+      (c.is_ascii_alphabetic() && c.is_ascii_lowercase()) || c.is_ascii_digit() || c == '_'
```

Comment:
Do you need the `is_ascii_alphabetic()` check?  I think `c.is_ascii_alphabetic() && c.is_ascii_lowercase()` == `c.is_ascii_lowercase()` for every `c`.

Reply:
good point, I'll simplify it